### ISSUE TITLE
Not escaping string in an alter password command

### ISSUE THREAD

**Issue by b-ryan** [lib/pq#708]
I am trying to run an alter user command with a dynamic password. code:

```go
package main

import (
	"database/sql"
	"log"

	_ "github.com/lib/pq"
)

func main() {
	connStr := "user=buck password=password dbname=buck sslmode=disable"
	db, err := sql.Open("postgres", connStr)
	if err != nil {
		log.Fatal(err)
	}
	newPassword := "newpassword"
	_, err = db.Query("ALTER USER buck WITH ENCRYPTED PASSWORD $1", newPassword)
	if err != nil {
		log.Fatal(err)
	}
}
```

fails with

```
2018/02/02 21:26:56 pq: syntax error at or near "$1"
```

Now I know that at least one issue like this was opened: https://github.com/lib/pq/issues/694 but I had thought this issue would only affect identifiers. I know I have been able to bind strings in this statement in Python, but that could have also been the library helping out.

So my question is whether this is the same as #694 and if it is, is there any way for me to get around it other than just writing my own escaping function?

**Comment by mjibson** [comment]
I think it's the same issue. Postgres appears to not support placeholders for that. Good bet is anything that starts with `ALTER` won't work with placeholders.

**Comment by cbandy** [comment]
Correct, only statements that can be `PREPARE`d will accept placeholders.

**Comment by cbandy** [comment]
https://www.postgresql.org/docs/current/static/sql-prepare.html

**Comment by b-ryan** [comment]
Lame - I don't suppose the library has a built-in function for escaping strings properly does it?

**Comment by nickuraltsev** [comment]
I've recently submitted a PR that adds a function for escaping strings: #718

**Comment by arp242** [comment]
Looks like this was added in #852
